### ISSUE TITLE
Design for 0-RTT

### ISSUE THREAD

**Issue by martinthomson** [martinthomson/http2v2#11]
0-RTT is newer than h2.  As it stands, when 0-RTT is used, the default settings apply because the server doesn't have any chance to send SETTINGS.  That's decidedly suboptimal.

**Comment by wtarreau** [comment]
Regarding the suboptimal aspect I agree, but how is this any different from the same case where it's sent over a plaintext TCP connection ? I mean, it's the whole server settings mechanism that is suboptimal, but we need something. Default values could possibly be advertised by other means (DNS etc), but similarly we could imagine that clients could cache last seen settings and try to follow them on new connections as long as they don't break the spec. For example learning MAX_CONCURRENT_STREAMS this way can make sense for a client.

**Comment by martinthomson** [comment]
What I'm thinking is that the server could indicate that its setting also apply to 0-RTT.  MAX_CONCURRENT_STREAMS is a bad example here as clients have to plan to exceed this, but I guess that knowing about a very low limit would be helpful.  I'm more thinking about things like the table size, which defaults to a limit that clients could reasonably exceed.  And then there are cases where clients might want to exercise an extension that is signaled using settings.

This is new feature work that might be better suited to an extension rather than a revision to the main specification.

**Comment by wtarreau** [comment]
So in short you would like to be able to signal that the settings are persistent beyond the current connection, that's it ?

**Comment by martinthomson** [comment]
Yeah, something like that.  We might have to be careful not to persist everything though.  QUIC has a mix of parameters, some that can apply to the next connection and others that apply only to a single connection.  I don't know if h2 will grow that.
